### DISCUSSION TITLE
Solcast 50 API call 'free trial plan' ending? (10 API call max)

### DISCUSSION THREAD

**Discussion by kme77** [nathanmarlor/foxess_em#363] [Q&A]
Has anyone else received an email saying the Solcast free tier will be moving to a paid model from 16th April? If the cost is prohibitive (i.e. costs more than we would save using it) is there an alternative or would the solar forecast portion of the integration need to be removed/deactivated?

**Comment by FozzieUK** [comment]
Yes also received the email (I have the older 50 API call account), the way the email reads, it seems that if you have an older account (or free trial as they call it) it will stop working on the 16th unless you email them and ask to convert it to the newer 'hobby' account (10 API calls), I have emailed them for clarification on that.

Sadly I don't know of any better forecast sites, maybe you can dedicate EM to using the 10 API solcast calls, and use the built in HA solar forecast for other tasks ?.

**Reply by kme77** [reply]
Yeah that's what I took from their email. I currently have the legacy 50 API call limit too but thought that with 10 API calls the integration runs out early on in the day so thought it wasn't useable with the 10 call limit. Do you know if it's possible to reduce the API calls to 10/day spread out evenly? I'm assuming that as 50 isn't available now that is the only workaround?

**Reply by jamesog** [reply]
FWIW I'm on a newer Solcast account that only gets 10 API calls a day and it's... painful. Those calls get used up quickly (I have two roofs and I think Energy Management can only call 2 or 3 times a day). This means the forecasts I see during the day are generally inaccurate.

I've been finding Solcast's forecasts to be wildly optimistic on cloudy days and I find myself having to micromanage what EM wants to do (usually adding some boost or disabling it entirely and add custom charge schedules). The [Forecast.solar](https://www.home-assistant.io/integrations/forecast_solar/) integration usually seems a bit more accurate, but still not perfect. I haven't been able to figure out the code in EM to be able to make it use the data from that integration instead of calling Solcast.

**Reply by FozzieUK** [reply]
It does work with 10 calls, it paces itself depending on the number of API call that solcast report as available starting from 6am.

The problem is it uses 2 calls each time it gets the forecasts, so that would be 5 times in total during the day - not brilliant, but still workable.

**Reply by FozzieUK** [reply]
and as @jamesog said, if you have multiple roofs then you'll only get (5 / number of roofs) during the day... very painful.

**Reply by kme77** [reply]
'Fortunately' I only have the one roof so I'd at least get the full 10 calls, though as you say it's not ideal for an accurate forecast. 

I'm wondering if the forecast NEEDS to be called throughout the day, as it's creating the charge schedule for the day after, based on the next days weather then would it not work just as well if the 10 API calls were used towards the end of the day (mimicking the last 10 calls of the 50)? As long as the data is called before the charge window (and later data will be more accurate) surely it will work jast as well as with 50, or am I misunderstanding how it works?

**Reply by jamesog** [reply]
It already calls the Solcast API 5 minutes before the eco start time, so you get a refresh then. The problems I've had are: 1) the new forecast is often >10 hours after the last call and can be very different, so sometimes EM does things I don't expect based on data earlier in the day; 2) even that call right before the start time often still isn't accurate enough leading to it thinking I'll generate more solar than I actually do 😞 

Of course on perfectly clear, sunny days it works just fine 😆

**Reply by kme77** [reply]
Yeah so that's why I'm thinking if the calls are ALL biased towards the end of the day then there should in theory be less divergence between data sets than if they're all hours apart, hopefully creating less 'odd' behaviour? To be honest, I've found with the 50 calls my integration has performed perfectly, though I DID edit my information in Solcast so that the predictions more closely matched the output I was seeing at home.

**Comment by kme77** [comment]
Had a reply from Solcast-

"Thank you for reaching out and using Solcast.

If you are currently using Solcast data for your home residential PV system, then this will not be changing on your account.

We will only be removing the following plans if they are on your account (of which should not impact your existing home residential PV hobbyist plan):
Monthly Averages Free
Timeseries Free
TMY Free

Apologies if there is any confusion."

I've replied asking for confirmation of legacy API call limits will remain at 50 or be reducing to 10 and also (cheekily) requested if it would be possible to increase the newer reduced limit to 50 during their payment restructuring. (Doesn't hurt to ask!) 😁

**Reply by FozzieUK** [reply]
Also had the same email - I did ask about the 50 api calls but they didn't specifically mention it so was going to email back; i'll not bother  now - would you let us know please if you get a response 👍

**Reply by kme77** [reply]
Sure, I'll update when I get a reply. 👍

**Reply by kme77** [reply]
Had a reply-

For the time being, we won't be changing those with 50 api call limits, as they were our early adopters, and we want to maintain that as a thank you.

As the adoption of solar panels rise, we've been seeing large increases of our hobbyist users, and so in order to be able to provide this data at no cost, we see that the 10 API calls are sustainable so that we'll be able to continue providing this service long into the future.

I am currently investigating the feasibility of a paid subscription plan in the future and I'd like to make a business case for it.

If you are interested, can you please fill out this form, and should we look to offer anything in the future, I'll be sure to let you know.

Link to form-
https://docs.google.com/forms/d/e/1FAIpQLSefZqRZ9z2f9pjl4Tc3zpscPVSBzhsPbErdyAwzui2xXqRz4A/viewform?usp=sf_link

**Reply by FozzieUK** [reply]
Thats good news, thanks 👍
